specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json

```diff
@@ -3790,60 +3773,197 @@
-      "x-ms-discriminator-value": "Filter"
+      "x-ms-discriminator-value": "MicrosoftSecurityIncidentCreation"
     },
-    "FilterAlertRuleTemplateProperties": {
-      "description": "Filter alert rule template property bag.",
+    "MicrosoftSecurityIncidentCreationAlertRuleCommonProperties": {
+      "description": "MicrosoftSecurityIncidentCreation rule common property bag.",
       "properties": {
-        "filterProduct": {
-          "description": "The filter product name for this template rule.",
-          "type": "string"
+        "productFilter": {
+          "description": "The alerts' productName on which the cases will be generated",
+          "enum":[
+          	"Microsoft Cloud App Security",
+          	"Azure Security Center",
+          	"Azure Advanced Threat Protection",
+          	"Azure Active Directory Identity Protection"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+          	"modelAsString": true,
+          	"name": "MicrosoftSecurityProductName"
+          }
         },
-        "filterSeverities": {
-          "description": "the alert’s severities on which the cases will be generated",
+        "severitiesFilter": {
+          "description": "the alerts' severities on which the cases will be generated",
           "items": {
             "$ref": "#/definitions/AlertSeverity"
           },
           "type": "array"
         },
-        "filterTitles": {
-          "description": "the alert’s titles on which the cases will be generated",
+        "displayNamesFilter": {
+          "description": "the alerts' displayNames on which the cases will be generated",
           "items": {
             "type": "string"
           },
           "type": "array"
         }
       },
+      "required": [
+      	"productFilter"
+      ],
+      "type": "object"
+    },
+    "FusionAlertRule": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/AlertRule"
+        }
+      ],
+      "description": "Represents Fusion alert rule.",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/FusionAlertRuleProperties",
+          "description": "Fusion alert rule properties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "type": "object",
+      "x-ms-discriminator-value": "Fusion"
+		},
+    "FusionAlertRuleProperties": {
+      "description": "Fusion alert rule base property bag.",
+      "properties": {
+        "alertRuleTemplateName": {
+          "description": "The Name of the alert rule template used to create this rule.",
+          "type": "string"
+        },
+        "description": {
+          "description": "The description of the alert rule.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "displayName": {
+          "description": "The display name for alerts created by this alert rule.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "enabled": {
```

Comment:
Booleans are fairly limiting.  If a rule could have more than an off/on state in the future it may be better to model this as an enum with "Enabled" and "Disabled" values that you can add to if necessary.

Reply:
I know. this 'enabled' field has only on/off state, and we don't change it now so we can keep our backwards compatibility.